core-bom/build.gradle

```diff
@@ -15,9 +15,33 @@ micronautBom {
 
 micronautBuild {
     binaryCompatibility {
-        def dash = version.indexOf('-')
-        def v = dash > 0 ? version.substring(0, dash) : version
-        def (major, minor, patch) = v.split('[.]').collect { it.toInteger() }
-        enabled = major > 4 || (major == 4 && minor > 0) || (major == 4 && minor == 0 && patch > 0)
+        baselineVersion = "4.0.0-RC3"
```

Comment:
doe we need to set the baseline version?

Reply:
https://github.com/micronaut-projects/micronaut-core/issues/9500#issuecomment-1609603337